library/src/main/java/com/firebase/ui/FirebaseArray.java

```diff
@@ -64,6 +107,24 @@ public DataSnapshot getItem(int index) {
         return mSnapshots.get(index);
     }
 
+    private void setup() {
+        if(mQuery != null) {
+            cleanup();
+        }
+        if(mPageSize == 0) {
+            mQuery = mOriginalQuery;
+        }
+        else if(mOrderASC == true){
+            mQuery = mOriginalQuery.limitToFirst(mCurrentSize);
+        }
+        else {
+            mQuery = mOriginalQuery.limitToLast(mCurrentSize);
+        }
+        setSyncing(true);
+        mQuery.addChildEventListener(this);
+        mQuery.addListenerForSingleValueEvent(this);
```

Comment:
As you scroll, this keeps reloading the pervious items for each next page. This is a bandwidth concern, which unfortunately I think can only be solved by loading each "page" separately and then merging them into a single array "under the hood".